public/react/Configure/components/Teams/services/team-configuration/team-configuration/fetcher.ts

```diff
@@ -29,5 +29,34 @@ export const getTeamData = (teamName: string, allTeamData: AllTeamData): TeamCon
 };
 
 export const updateAllTeamData = async (storage: AppStorage, updates: AllTeamData): Promise<void> => {
-  return storage.update(allTeamDataKey, updates);
+  const sanitized = entriesFlatMap(
+    updates,
+    ([teamConfigKey, updatedTeamConfig]: [string, TeamConfiguration | undefined]) => {
+      if (!updatedTeamConfig) return [];
+
+      const value = entriesFlatMap(
+        updatedTeamConfig,
+        ([configKey, updatedConfig]: [string, Configuration | undefined]) => {
+          if (!updatedConfig) return [];
+
+          const configNoNulls = filterNullValues(updatedConfig);
+          
+          return notEmpty(configNoNulls, [configKey, configNoNulls]);
+        }
+      );
+      return notEmpty(value, [teamConfigKey, value]);
+    }
+  );
+  return storage.update(allTeamDataKey, sanitized);
 };
+
+const entriesFlatMap = <TItem, TMap extends Record<string, TItem>>(
+  updates: TMap,
+  fn: ([key, update]: [string, TItem]) => (string | { [k: string]: any })[][]
```

Comment:
what is up with the typing of the return of this? we should know all the shapes.